src/Common/src/Interop/Windows/NtDll/Interop.RtlGetVersion.cs

```diff
@@ -20,8 +19,19 @@ internal static unsafe string RtlGetVersion()
             const string version = "Microsoft Windows";
             if (RtlGetVersion(ref osvi) == 0)
             {
-                return string.Format("{0} {1}.{2}.{3} {4}",
-                    version, osvi.dwMajorVersion, osvi.dwMinorVersion, osvi.dwBuildNumber, new string(&(osvi.szCSDVersion[0])));
+                var stringBuilder = new StringBuilder();
+
+                stringBuilder.Append(version);
+                stringBuilder.Append(" ");
+                stringBuilder.AppendFormat("{0}.{1}.{2}", osvi.dwMajorVersion, osvi.dwMinorVersion, osvi.dwBuildNumber);
+
+                if (osvi.szCSDVersion[0] != '\0')
+                {
+                    stringBuilder.Append(" ");
+                    stringBuilder.Append(new string(&(osvi.szCSDVersion[0])));
+                }
+
+                return stringBuilder.ToString();
```

Comment:
How about just changing the existing:
```C#
return string.Format("{0} {1}.{2}.{3} {4}",
    version, osvi.dwMajorVersion, osvi.dwMinorVersion, osvi.dwBuildNumber, new string(&(osvi.szCSDVersion[0])));
```
to instead be:
```C#
return string.Format("{0} {1}.{2}.{3}{4}",
    version,
    osvi.dwMajorVersion,
    osvi.dwMinorVersion,
    osvi.dwBuildNumber,
    osvi.szCSDVersion[0] != '\0' ? " " + new string(&(osvi.szCSDVersion[0])) : string.Empty);
```
?

Reply:
I wrote exactly this code but wanted to check with someone on the .NET maintainers list. I spoke to Viktor Hofer who asked that I consider using a `StringBuilder` instead.

Reply:
To be clear I recommended to either use string.Format, StringBuilder or Span if its a hot path :)